### PR TITLE
GTW-1106: Support Retry Policy To SQS

### PR DESCRIPTION
**Pull request by konstantinos-kafkas**
### Jira link
-  [GTW-1106](https://checkout.atlassian.net/browse/GTW-1106)

### Technical comments
- SqsConnectionRetryCount setting added -> retry connection count
- SqsConnectionRetryBackoffMultiplier setting added -> backoff multiplier

### Functional comments
peekaqueue should be resilient to network issues when is trying to connect to SQS. Retry policy is implemented such that if an exception is thrown during the GetQueueURL it retries. 

### How to test
- Turn off the alpine-sqs
- Run peekaqueue
- On the second retry turn on alpine-sqs
- Check the console for the retry logs 

[Click here for the PR help checklist](https://checkout.atlassian.net/wiki/spaces/GW/pages/385876274/PR+checklist)

### CODE REVIEW

**Comment by konstantinos-kafkas** [comment]
@ben-foster-cko @benfoster  ↑↑↑

**Review by ben-foster-cko — CHANGES_REQUESTED** [review]
Thanks for the contribution @konstantinos-kafkas-cko!

As discussed, I'd like to handle the queue retrieval more gracefully. You can change the `GetQueuesToMonitor()` to execute the retry policy and disregard any queues that can be loaded so we don't kill the app if just one out of x queues fail.

Secondly, it makes sense to wrap all SQS calls in a retry policy since any of them could fail for one reason or another.

Finally, if no queues can be resolved, you can inject `IApplicationLifetime` and shutdown the application (make sure you add a warning to this effect).

File: src/Peekaqueue/SqsStatsService.cs [hunk 1]
```diff
@@ -52,20 +61,24 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             {
                 var queues = await GetQueuesToMonitor();
 
-                if (queues.Count == 0)
+                if (queues.Count > 0)
```
Comment (ben-foster-cko) [inline src/Peekaqueue/SqsStatsService.cs]: You can reverse this logic to `queues.Count == 0`, exit early and avoid the nesting

Comment (konstantinos-kafkas) [reply src/Peekaqueue/SqsStatsService.cs]: Done

**Review by ben-foster-cko — APPROVED** [review]

**Pull request merged**